projects/libs/core-library-typescript/src/common-library/block-utils.ts

```diff
@@ -14,23 +20,60 @@ export namespace BlockUtils {
     return hash.digest();
   }
 
-  export function buildBlock(blockComponents: {header: types.BlockHeader, body: types.BlockBody}): types.Block {
+  export function buildBlock(blockComponents: {header: types.BlockHeader, body: types.BlockBody}, options?: BlockUtilsConfig): types.Block {
```

Comment:
I think it's wrong to pass all this config to BlockUtils.  It's enough, IMO, to add an optional key class that can be used to sign a block